### PR TITLE
adding rangefilter to the getOMMemoryRegions gating expression

### PR DESCRIPTION
**Pull request by derekpappas**
**Related issue**: 

**Type of change**: bug report

**Impact**: API modification

**Development Phase**:  implementation

**Release Notes**

Added an additional gating condition to the  getOMMemoryRegions memory region creation function. This condition fixes a bug where certain memories were not included in the memory regions.

### CODE REVIEW

File: src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala [hunk 1]
```diff
@@ -168,7 +168,7 @@ object DiplomaticObjectModelAddressing {
 
   def getOMMemoryRegions(name: String, resourceBindings: ResourceBindings, omRegMap: Option[OMRegisterMap] = None): Seq[OMMemoryRegion]= {
     resourceBindings.map.collect {
-      case (x: String, seq: Seq[Binding]) if (DiplomacyUtils.regFilter(x)) =>
+      case (x: String, seq: Seq[Binding]) if (DiplomacyUtils.regFilter(x)) || (DiplomacyUtils.rangeFilter(x))=>
```
Comment (richardxia) [inline src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala]: Nit: Put a space before the `=>` arrow.
Nit 2: I think the parentheses are unnecessary. Does the following suggestion compile?

```suggestion
      case (x: String, seq: Seq[Binding]) if DiplomacyUtils.regFilter(x) || DiplomacyUtils.rangeFilter(x) =>
```

Comment (derekpappas) [reply src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala]: Does the gating expression in getOMPortMemoryRegions need the same set of conditions?

Comment (richardxia) [reply src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala]: Oh, sorry, I didn't read this PR carefully enough and mixed it up with the other one. This change should *only* go on the ports, although it's probably not too harmful to put it on the memories.

Comment (derekpappas) [reply src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala]: Added

**Pull request merged**